server/src/main/java/org/elasticsearch/index/translog/TranslogReader.java

```diff
@@ -70,6 +75,30 @@ public static TranslogReader open(
         return new TranslogReader(checkpoint, channel, path, header);
     }
 
+    /**
+     * Closes current reader and creates new one with new checkoint and same file channel
+     */
+    TranslogReader closeIntoTrimmedReader(long belowTerm, long aboveSeqNo, ChannelFactory channelFactory) throws IOException {
+        if (!closed.get()
```

Comment:
I think we shouldn't be lenient here. Always close the reader and adhere to the a simple API - i.e., the current reader is always closed, a new reader is always returned. Also if the current reader is already closed, you can throw an AlreadyClosedException (it's wrong to be lenient about it - it hides bugs).